CONTRIBUTING.md

```diff
@@ -0,0 +1,618 @@
+
+## GitHub basic setup
+
+To get started, complete the prerequisites below.
+
+### Prerequisites
+
+- Review [Contributor roles](#contributor-roles) that require special Git 
+  configuration.
+
+- [Set up Git](https://help.github.com/en/github/getting-started-with-github/set-up-git). 
+  
+  >**Note:** The email address you specify must match the email address you 
+  >use to sign-off commits.
+
+- [Fork and Clone](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) the relevant repository at the 
+  [Raksh Project](https://github.com/IBM/raksh).
+
+   Example: Your local clone should show `your-github-username`, as follows.
+   `https://github.com/${your-github-username}/community`.
+
+### Contributor roles
+
+Special Git configuration is required for these contributors:
+
+* [Golang coding style](#golang-coding-style)
+
+For all other contributor roles, follow the standard configuration, shown in
+Prerequisites. 
+
+### Golang coding style
+
+* Review [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments) to avoid common `Golang` errors.
+* Use `gofmt` to fix any mechanical style issues.
+
+### Dependency management
+In order to add or update a dependency to this project run:
+```
+> export GO111MODULE=on
+> go get -u [DEPENDENCY]
+```
+Since RAKSH uses go modules we highly recommend reading the [go modules
+wiki](https://github.com/golang/go/wiki/Modules), especially the [daily workflow
+section](https://github.com/golang/go/wiki/Modules#daily-workflow).
+To ensure the working directory contains all necessary files afterwards, run:
+```
+> make vendor
+```
+
+## Certificate of Origin
+
+In order to get a clear contribution chain of trust we use the [signed-off-by
+language](https://ltsi.linuxfoundation.org/software/signed-off-process/)
+used by the Linux kernel project.
+
+## GitHub best practices
+
+### Submit issues before PRs
+
+Raise a GitHub issue **before** starting work on a PR.
+
+* Our process requires an issue to be associated with every PR (see [patch format](#patch-format))
+
+* If you are a new contributor, create an issue and add a comment stating 
+that you intend to work on the issue. This notifies our team of the work you 
+plan to do.
+
+### Issue tracking
+
+To report a bug that is not already documented, please open a GitHub issue for the repository in question.
+
+If it is unclear which repository to raise your query against, first try to
+get in [contact](#contact) with us. If in doubt, raise the issue
+[here](https://github.com/IBM/raksh/issues/new) and we will
+help you to handle the query by routing it to the correct area for resolution.
+
+### Closing issues
+
+Our tooling requires adding a `Fixes` comment to at least one commit in the PR, which triggers GitHub to automatically close the issue once the PR is merged:
+
+```
+pod: Remove token from Cmd structure
```

Comment:
For now we can put it simply as `Remove token from Cmd structure` without the subsystem.